L1Trigger/L1TGlobal/src/GlobalBoard.cc

```diff
@@ -973,12 +973,14 @@ void l1t::GlobalBoard::runFDL(edm::Event& iEvent,
     // prescale counters: numberPhysTriggers counters per bunch cross
     m_prescaleCounterAlgoTrig.reserve(totalBxInEvent);
 
-    auto const& prescaleCountersAlgoTrig =
-        m_semiRandomInitialPSCounters ? semirandomNumber(iEvent, prescaleFactorsAlgoTrig) : prescaleFactorsAlgoTrig;
-
     for (int iBxInEvent = 0; iBxInEvent <= totalBxInEvent; ++iBxInEvent) {
```

Comment:
I haven't changed any logic other than that necessary for the fractional prescales implementation. An expert should comment here.

Reply:
This is definitely worth being investigated, even if not in this PR:
Moreover, the whole chunk of code in [L972-L998](https://github.com/cms-sw/cmssw/blob/563f3fbbe17566cc12fd08659c47079e3132bbc3/L1Trigger/L1TGlobal/src/GlobalBoard.cc#L972-L998) can be highly simplified and improved, in order to avoid running twice the loop in the second `if` block

Reply:
I am not sure what the difference is between m_firstEv and m_firstEvLumiSegment. It does seem like there may be some redundancy, which existed before this pull request. I did not make changes to the logic here. Is there someone to contact who can give more input?

Reply:
Reading the code, I think `m_firstEv` and `m_firstEvLumiSegment` have a similar function, but are split into two values because the second part (the one protected by `m_firstEvLumiSegment`) can be triggered also by a different condition.

It could be simplified using `m_firstEv` in both places and moving `m_firstEv = false` outside of the checks, after the second one.

Reply:
Also about `totalBxInEvent`, from reading the code I agree that it should be OK to make the loop go from `0` to `totalBxInEvent - 1`.

However, the loop is like that since the first implementation 7 years ago... so I doubt anybody remembers the reason now :-D

I'd just try changing `iBxInEvent <= totalBxInEvent` to `iBxInEvent < totalBxInEvent`, test on both MC and data, and see what happens.

Reply:
Okay, let's postpone this to a separate PR.

Reply:
I'm going to mark this as "resolved" in this PR to postpone the discussion to another PR. I moved the discussion to an issue https://github.com/cms-sw/cmssw/issues/39817.